### PR TITLE
Added Staff Shift endpoints

### PR DESCRIPTION
**Pull request by akulsharma1**
- added all staff shift endpoints
- NOTE: `/staff/shift` returns dummy data at the moment, we are working on bugfixing a timeout request issue but wanted to push out an MVP.

### CODE REVIEW

File: src/services/staff/staff-formats.ts [hunk 1]
```diff
@@ -3,6 +3,43 @@ export interface AttendanceFormat {
...
+    isAsync: boolean;
+}
+
+// Function to generate a single event entry
+export function generateEvent(eventId: number): EventData {
+    return {
+        eventId: "event " + eventId,
+        name: "test " + eventId,
+        description: "...",
+        startTime: 1696467600,
```
Comment (lasyaneti) [inline src/services/staff/staff-formats.ts]: could you set these to be during hackillinois feb 23-26 2024 since mobile only displays those dates

**Review by lasyaneti — APPROVED** [review]
LGTM for now. Just change dates for staff shifts (might need to do it manually in mongo for existing events too) and we can merge.

Eventually `GET /staff/shift` should only return events for authenticated staff.

File: src/services/event/event-router.ts [hunk 2]
```diff
@@ -64,15 +64,26 @@ eventsRouter.use(cors({ origin: "*" }));
...
+eventsRouter.get("/staff/", strongJwtVerification, async (req: Request, res: Response, next: NextFunction) => {
     const payload: JwtPayload = res.locals.payload as JwtPayload;
 
     if (!hasStaffPerms(payload)) {
         return next(new RouterError(StatusCode.ClientErrorForbidden, "Forbidden"));
     }
 
-    const staffEvents: StaffEvent[] = await Models.StaffEvent.find();
-    return res.status(StatusCode.SuccessOK).send({ events: staffEvents });
+    const showAllShiftsParam: string | undefined = req.query.showAllShifts as string | undefined;
```
Comment (lasyaneti) [inline src/services/event/event-router.ts]: also im only testing with my token (admin) but make sure that if non-admin staff roles request showAllShifts=True, it redirects them to just shifts assigned to them - they shouldnt be able to see everything

File: src/services/event/event-router.ts [hunk 3]
```diff
@@ -64,15 +64,26 @@ eventsRouter.use(cors({ origin: "*" }));
...
+    }
+
+    if (showAllShifts) {
+        const staffEvents: StaffEvent[] = await Models.StaffEvent.find();
+        return res.status(StatusCode.SuccessOK).send({ events: staffEvents });
+    } else {
+        const staffEvents: StaffEvent[] = await Models.StaffEvent.find({ eventType: { $ne: "STAFFSHIFT" } });
+        return res.status(StatusCode.SuccessOK).send({ events: staffEvents });
+    }
 });
```
Comment (lasyaneti) [inline src/services/event/event-router.ts]: should be protected under admin perms

-aydan (from lasya's computer)

Comment (lasyaneti) [reply src/services/event/event-router.ts]: add a check for admin perms in jwt

File: src/services/staff/staff-router.ts [hunk 4]
```diff
@@ -76,4 +77,41 @@ staffRouter.post("/attendance/", strongJwtVerification, async (req: Request, res
...
+        const staffEvents: StaffEvent[] = await Models.StaffEvent.find({eventId: {$all: events}});
+        return res.status(200).json(staffEvents);
+        */
+
+        const eventArray: EventData[] = [];
+        const tries: number = 5;
+        for (let i = 1; i <= tries; i++) {
+            const e: EventData = generateEvent(i);
+            eventArray.push(e);
+        }
```
Comment (lasyaneti) [inline src/services/staff/staff-router.ts]: return hardcoded json instead

Comment (lasyaneti) [reply src/services/staff/staff-router.ts]: no need to run function each time

**Review by lasyaneti — CHANGES_REQUESTED** [review]

File: src/services/event/event-router.ts [hunk 5]
```diff
@@ -64,15 +64,26 @@ eventsRouter.use(cors({ origin: "*" }));
  * @apiUse strongVerifyErrors
  * @apiError (403: Forbidden) {String} Forbidden Not a valid staff token.
  * */
-eventsRouter.get("/staff/", strongJwtVerification, async (_: Request, res: Response, next: NextFunction) => {
+eventsRouter.get("/staff/", strongJwtVerification, async (req: Request, res: Response, next: NextFunction) => {
     const payload: JwtPayload = res.locals.payload as JwtPayload;
 
-    if (!hasStaffPerms(payload)) {
+    if (!hasAdminPerms(payload)) {
```
Comment (AydanPirani) [inline src/services/event/event-router.ts:111]: Why change this? All staff should be able to see staff events

Hierarchy should be like this:

non-staff: no access
staff: access to their own shifts + all general staff events
admins: access to staff shifts

**Pull request closed without merging**